RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h

```diff
@@ -1,240 +1,251 @@
+      pi_ = &pi.front();
+    }
+    
+    double * z_; // z-coordinate at point of closest approach to the beamline
+    double * t_; // t-coordinate at point of closest approach to the beamline
+    double * pi_; // track weight
+
+    double * dz2_; // square of the error of z(pca)
+    double * dt2_; // square of the error of t(pca)
+    double * Z_sum_; // Z[i]   for DA clustering
+    
+    std::vector<double> z; // z-coordinate at point of closest approach to the beamline
+    std::vector<double> t; // t-coordinate at point of closest approach to the beamline
+    std::vector<double> dz2; // square of the error of z(pca)
+    std::vector<double> dt2; // square of the error of t(pca)
+    std::vector<double> Z_sum; // Z[i]   for DA clustering
+    std::vector<double> pi; // track weight
+    std::vector< const reco::TransientTrack* > tt; // a pointer to the Transient Track
+  };
+  
+  struct vertex_t {
+    
+    void addItem( double new_z, double new_t, double new_pk   )
+    {
+      z.push_back( new_z);
+      t.push_back( new_t);
+      pk.push_back( new_pk);
+      
+      ei_cache.push_back( 0.0 );
+      ei.push_back( 0.0 );
+      swz.push_back( 0.0);
+      swt.push_back( 0.0);
+      se.push_back( 0.0);
+      nuz.push_back(0.0);
+      nut.push_back(0.0);
+      szz.push_back(0.0);
+      stt.push_back(0.0);
+      szt.push_back(0.0);
+      
+      extractRaw();
+    }
+    
+    unsigned int getSize() const
+    {
+      return z.size();
+    }
+
+    // has to be called everytime the items are modified
+    void extractRaw()
+    {
+      z_ = &z.front();
+      t_ = &t.front();
+      pk_ = &pk.front();
+      
+      ei_ = &ei.front();
+      swz_ = &swz.front();
+      swt_ = &swt.front();
+      se_ = &se.front();
+      nuz_ = &nuz.front();
+      nut_ = &nut.front();
+      szz_ = &szz.front();
+      stt_ = &stt.front();
+      szt_ = &szt.front();
+
+      ei_cache_ = &ei_cache.front();
+      
+    }
+
+    void insertItem( unsigned int i, double new_z, double new_t, double new_pk   )
+    {
+      z.insert(z.begin() + i, new_z);
+      t.insert(t.begin() + i, new_t);
+      pk.insert(pk.begin() + i, new_pk);
+      
+      ei_cache.insert(ei_cache.begin() + i, 0.0 );
+      ei.insert( ei.begin()  + i, 0.0 );
+      swz.insert(swz.begin() + i, 0.0 );
+      swt.insert(swt.begin() + i, 0.0 );
+      se.insert( se.begin()  + i, 0.0 );
+      nuz.insert(nuz.begin() +i, 0.0 );
+      nut.insert(nut.begin() +i, 0.0 );
+      szz.insert(szz.begin() + i, 0.0 );
+      stt.insert(stt.begin() + i, 0.0 );
+      szt.insert(szt.begin() + i, 0.0 );
+      extractRaw();
+    }
+    
+    void removeItem( unsigned int i )
+    {
+      z.erase( z.begin() + i );
+      t.erase( t.begin() + i );
+      pk.erase( pk.begin() + i );
+      
+      ei_cache.erase( ei_cache.begin() + i);
+      ei.erase( ei.begin() + i);
+      swz.erase( swz.begin() + i);
+      swt.erase( swt.begin() + i);
+      se.erase(se.begin() + i);

```

Comment:
shouldn't you erase also the new vectors nux, nut, szz, stt, szt?

Reply:
I suppose in the end it does not matter since the indices will simply be overwritten (as the length of all the vectors is kept track of by the length of z). However, you are definitely correct that we should be consistent here!